contracts/exchange/Orderbook.sol

```diff
@@ -139,6 +144,40 @@ contract Orderbook is IOrderbook, ManagerBase {
                 orderAmounts[i] = orders[_orderIds[i]].amountOrdered - orders[_orderIds[i]].amountFilled;
             }
         }
+
+        // get amounts ordered based on AmountToFill and Max Spend
+        amountFilled = 0;
+        uint256 amountSpentOnOrder = 0;
+        for (uint256 i = 0; i < orderAmounts.length; ++i) {
+            if (orderAmounts[i] > 0) {
+                amountSpentOnOrder = orders[_orderIds[i]].price * orderAmounts[i];
+                
+                // Check if there's you're still under tha max Spend
+                if (maxSpend == 0) {
+                    orderAmounts[i] = 0;
+                    continue;
+                } else if (maxSpend > amountSpentOnOrder) {
+                    maxSpend -= amountSpentOnOrder;
+                } else if (maxSpend < amountSpentOnOrder) {
+                    orderAmounts[i] = maxSpend / orders[_orderIds[i]].price;
```

Comment:
Is it okay for orderAmounts[i] to store a non-whole number?

Reply:
Oh I see it's uint256 so now let me rephrase the question, is it okay for it to truncate the decimal in the event there is one?

Reply:
We want to round down here. 

I.e. if max spend is 100, and price is 30, we want the orderAmount for this index to round down to 3 items.

Reply:
Note that, this assumes that the orders are in price ascending / descending order (depending on buy/sell). So in the example above, 30 is the lowest possible price and that's why we're setting maxSpend = 0. The next order _should_ be equal or higher than price 30.